### PR TITLE
Groceries Store. Homework with collections.

### PR DESCRIPTION
**Pull request by Cherkashyn**
Hey! Accept my homework!

### CODE REVIEW

File: homework1/src/main/java/lesson6/Food.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package lesson6;
+
+import java.time.LocalDate;
+
+public interface Food {
+    Type getType();
+    LocalDate getExpirationDate();
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson6/Food.java]: You use expiration date. If you use it way you must cheak that expirationDate>deliveryDate. It not critical

File: homework1/src/main/java/lesson6/Food.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package lesson6;
+
+import java.time.LocalDate;
+
+public interface Food {
+    Type getType();
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson6/Food.java]: You use enum Type for fruits. What you will be doing if you have Meat that extends Food? =)

File: homework1/src/main/java/lesson6/Groceries.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+public class Groceries {
+    private List<Food> groceriesList;
+
+    public Groceries(Food... food) {
+        groceriesList = Arrays.asList(food);
+    }
+
+    // just printed all items in a list
+    public void print(){
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson6/Groceries.java]: Don't need. Use toString

File: homework1/src/main/java/lesson6/Groceries.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        System.out.println(groceriesList.toString());
+    }
+
+    //returns all fresh fruits up to current date
+    public void allFresh(LocalDate currentDate){
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson6/Groceries.java]: In this methods better way return list. You can reuse this method in others. Change also in others

Comment (Cherkashyn) [reply homework1/src/main/java/lesson6/Groceries.java]: ok

File: homework1/src/main/java/lesson6/Groceries.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        List<Food> tempList = new ArrayList<>();
+
+        for (Food food : groceriesList){
+            if (food.getExpirationDate().isAfter(currentDate)){
+                tempList.add(food);
+            }
+        }
+
+        System.out.println("____________________________________________________________");
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson6/Groceries.java]: There are must present only logic. Sysout use in main class

Comment (SergeyPasko) [reply homework1/src/main/java/lesson6/Groceries.java]: you many time repeat same logic to Sysout

File: homework1/src/main/java/lesson6/Groceries.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            if (food.getType().equals(type)){
+                tempList.add(food);
+            }
+        }
+
+        return tempList;
+    }
+
+    //returns food's price with discount
+    public Map<Food, Double> discount(Type type, long remainDays, double discount){
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson6/Groceries.java]: Read homeTask. I wriiten about "Якщо срок реалізації менше певного часу (параметр) то змінити ціну (параметр процент), але це стосується тільки певних видів фруктів". It is mean than types in params can be many. List\<Types\> or Types[] or Types... You have method with "param...". Now I say -ok

File: homework1/src/main/java/lesson6/Groceries.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        Map<Food, Double> tempMap = new HashMap<>();
+
+        for (Food food : onlySpecialTypeWithStorageTimeMoreThan(type, remainDays)){
+            food.setPrice((food.getPrice() * discount));
+            tempMap.put(food, food.getPrice());
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson6/Groceries.java]: You put to map key-value. Key is mutable class. You can do this, but in future you can have problems

**Pull request merged**